gateway/api.go

```diff
@@ -1508,6 +1508,10 @@ func createOauthClient(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		if apiSpec.Oauth2Meta.RequirePKCE {
+			newClient.ClientSecret = ""
+		}
+
```

Comment:
Why are we forcing ClientSecret to be empty? What if the client is a confidential client - meaning that it can send the client secret in the authorization request?

Reply:
In that case that partilcular API would not have `RequirePKCE` set and would work without the code challenge

Reply:
What happens if one client is a public client, and the other is a confidential one? How to let both public and confidential clients use this API?

If we force PKCE, then confidential clients would send their client_secret and code_challenge in the authorize request. We should also validate the client_secret right?

Reply:
Thats a limitation for sure - if you want strict enforcement of pkce then you can only expose that API to public clients. Why you would want both at the same time on the same API is worth diving into though.